engine/src/org/pentaho/di/trans/steps/mergerows/MergeRows.java

```diff
@@ -257,7 +257,7 @@ public boolean init( StepMetaInterface smi, StepDataInterface sdi ) {
    * @throws KettleRowException
    *           in case there is a compatibility error.
    */
-  protected void checkInputLayoutValid( RowMetaInterface referenceRowMeta, RowMetaInterface compareRowMeta ) throws KettleRowException {
+  static void checkInputLayoutValid( RowMetaInterface referenceRowMeta, RowMetaInterface compareRowMeta ) throws KettleRowException {
```

Comment:
The `protected` was removed so that it can be referenced in the Meta class, and it was made `static` as the method does not make use of any state of the class, only the two arguments, so it didn't need to run in the context of the instance object.